services/core/shared/core/compat/sdk_v5/voters.js

```diff
@@ -34,22 +34,39 @@ const extractAddressFromPublicKey = pk => (getAddressFromPublicKey(Buffer.from(p
 
 const indexVotes = async blocks => {
 	const votesDB = await getVotesIndex();
+	const votesAggregateDB = await getVotesAggregatedIndex();
 	const votesMultiArray = blocks.map(block => {
 		const votesArray = block.payload
 			.filter(tx => tx.moduleID === dposModuleID && tx.assetID === voteTransactionAssetID)
 			.map(tx => {
 				const voteEntries = tx.asset.votes.map(vote => {
 					const voteEntry = {};
 
-					// TODO: Remove 'tempId' after composite PK support is added
-					voteEntry.tempId = tx.id.concat(vote.delegateAddress);
-					voteEntry.id = tx.id;
 					voteEntry.sentAddress = getBase32AddressFromHex(
 						extractAddressFromPublicKey(tx.senderPublicKey),
 					);
 					voteEntry.receivedAddress = getBase32AddressFromHex(vote.delegateAddress);
 					voteEntry.amount = vote.amount;
 					voteEntry.timestamp = block.timestamp;
+
+					// indexing aggregated votes per account
+					votesAggregateDB.increment({
+						increment: {
+							amount: BigInt(vote.amount),
+						},
+						where: {
+							property: 'id',
+							value: voteEntry.receivedAddress.concat(voteEntry.sentAddress),
+						},
+					}, {
+						...voteEntry,
+						id: voteEntry.receivedAddress.concat(voteEntry.sentAddress),
+					});
```

Comment:
Kindly consider adding an additional check to verify that the transaction containing the vote information isn't already indexed before performing any increment. This is necessary to avoid replays.

Alternatively, it might also be beneficial to add a flag within the original `votes` table that tracks if a particular vote transaction is already reflected within the `votes_aggregated` index.

Reply:
Added additional logic to verify that.